src/main/java/org/galatea/starter/object/StockDay.java

```diff
@@ -0,0 +1,142 @@
+package org.galatea.starter.object;
+
+import com.fasterxml.jackson.annotation.JsonAlias;
+import com.fasterxml.jackson.annotation.JsonIgnore;
+import com.fasterxml.jackson.annotation.JsonProperty;
+import com.fasterxml.jackson.databind.annotation.JsonDeserialize;
+import com.fasterxml.jackson.databind.annotation.JsonSerialize;
+import java.io.Serializable;
+import java.math.BigDecimal;
+import java.time.LocalDate;
+import java.time.LocalTime;
+import java.time.OffsetDateTime;
+import java.time.ZoneOffset;
+import java.util.List;
+import java.util.Map;
+import java.util.Objects;
+import javax.persistence.Column;
+import javax.persistence.Entity;
+import javax.persistence.Id;
+import javax.persistence.IdClass;
+import javax.persistence.Table;
+import lombok.AllArgsConstructor;
+import lombok.Builder;
+import lombok.Data;
+import lombok.RequiredArgsConstructor;
+import org.galatea.starter.service.object.AvResponse;
+import org.galatea.starter.utils.json.serialization.DateDeserializer;
+import org.galatea.starter.utils.json.serialization.DateSerializer;
+import org.galatea.starter.utils.json.serialization.MoneyDeserializer;
+
+@Data
+@Entity
+@Builder
+@Table(name = "price")
+@IdClass(StockSymbolId.class)
+@RequiredArgsConstructor
+@AllArgsConstructor
+public class StockDay implements Serializable, Comparable<StockDay> {
+
+  @Id
+  @JsonIgnore
+  private int stockId;
+  @JsonDeserialize(using = MoneyDeserializer.class)

```

Comment:
Don't think you need these @JsonDeserialize annotations because you are only ever serializing this object

Reply:
❔ the alphavantage request, deserializes into a map of these objects, using the JsonAlias-es

Reply:
Oh ok, I see now.  Using the StockDay object as representation of part of the AvResposne as well as how you are storing it internally and redisplaying it for the user seems to be a bit much.  

I'd suggest creating 2 classes to represent StockDay. One that you can put in your starter.service.object package whose purpose is to model the response from alphavantage (named AvStockDay or something). Then another in your starter.object package that would model how your app views a stock day internally (A good name for this package would be starter.domain).  

You can keep the both of the classes essentially identical at this point because it doesn't really matter for your app. And the translation from one to another may seem tedious, but decoupling those two things would help you making changes to your app in the future.

Reply:
Another small thing is if MoneyDeserializer is just deserializing to BigDecimal I think it shouldn't be necessary?   Because Feign uses GSON which I believe supports deserialization to BigDecimal. Let me know if I'm wrong on that

Reply:
✔separated these out to two separate classes and added some logic to convert AvStockDay -> StockDay. also, removed the money deserializer because you're right it does it just fine on it's own and i'm honestly not sure why i added that to begin with.